src/coreComponents/linearAlgebra/interfaces/hypre/SuperluUtils.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SuperluUtils.cpp
+ */
+
+#include "SuperluUtils.hpp"
+#include "common/Stopwatch.hpp"
+
+namespace geosx
+{
+
+void ConvertToSuperMatrix( HypreMatrix const & matrix,
+                           array1d< globalIndex > & rowPtr,
+                           array1d< globalIndex > & cols,
+                           array1d< real64 > & vals,
+                           SuperMatrix & SLUDMat )
+{
+  localIndex const numLocalRows = matrix.numLocalRows();
+  rowPtr.resize( numLocalRows+1 );
+  cols.reserve( matrix.numLocalNonzeros() );
+  vals.reserve( matrix.numLocalNonzeros() );
+  rowPtr[0] = 0;
+  for( localIndex i = 0; i < numLocalRows; ++i )
+  {
+    localIndex const nonZeros = matrix.localRowLength( i );
+    array1d< globalIndex > colIndices( nonZeros );
+    array1d< real64 > values( nonZeros );
+    matrix.getRowCopy( matrix.getGlobalRowID( i ), colIndices, values );
+    cols.insert( rowPtr[i], colIndices.begin(), colIndices.end() );
+    vals.insert( rowPtr[i], values.begin(), values.end() );
+    rowPtr[i+1] = rowPtr[i] + nonZeros;
+  }
```

Comment:
This could be done by constructing in `HypreSolver::solver_direct` a local matrix obtained by merging diagonal and off-diagonal part:

```
matrixLocal = hypre_MergeDiagAndOffd(matrix);
```

and passing to `ConvertToSuperMatrix` two parameters: `matrixLocal` and `superlu_data_struct` (the data structure mentioned above)

Reply:
You are right. Everything here has been designed to be portable across different interfaces. By focusing only on _hypre_ almost nothing will survive! I have to redesign the entire logic, but first of all, I want all the parameters exposed.